src/device.coffee

```diff
@@ -131,13 +132,13 @@ setBootConfig = (env, logMessage) ->
 				configStatement = null
 				if _.includes(toBeChanged, key)
 					configStatement = "#{key}=#{configFromApp[key]}"
-				else
+				else if !_.includes(toBeDeleted, key)
 					configStatement = configStatements[index]
 				return configStatement
 			# Here's the dangerous part:
 			execAsync("mount -t vfat -o remount,rw #{bootBlockDevice} #{bootMountPoint}")
 			.then ->
-				fs.writeFileAsync(bootConfigPath + '.new', outputConfig.join('\n'))
+				fs.writeFileAsync(bootConfigPath + '.new', _.without(outputConfig, null).join('\n'))
```

Comment:
`_.without(outputConfig, null)` will still allow undefined - you could use `_.reject(outputConfig, _.isNil)` to avoid that